dashboard/app/models/pd/workshop_survey.rb

```diff
@@ -127,15 +121,66 @@ def validate_required_fields
 
     # if this is the first survey completed by this user, also require
     # demographics questions
-    if pd_enrollment && self.class.find_by_user(pd_enrollment.user).empty?
-      demographics_required_fields.each do |field|
-        add_key_error(field) unless hash.key?(field)
+    if pd_enrollment
+      if pd_enrollment.user.nil? || self.class.find_by_user(pd_enrollment.user).empty?
```

Comment:
Is this the only place this logic is used? It might be nice to move this into a method, e.g. `first_survey_for_user?`, with a unit test.